src/assignments/benson/lesson_2_answers.clj

```diff
@@ -1,3 +1,59 @@
 (ns lesson-2-answers)
 
-;;; Enter your answers in here.
+;; Implements filter using a loop
+(defn filter-loop
+  [pred input]
+  ;; recurses through input sequence and conjoins all variables
+  ;; that pass the predicate condition
+  (loop [xs input
+         result []]
+    (if xs
+      (let [x (first xs)]
+        (if (pred x)
+          (recur (next xs) (conj result x))
+          (recur (next xs) result)))
+      result)))
+
+
+;; Implements filter using reduce
+(defn filter-reduce
+  [pred input]
+  (reduce
+   (fn [output x]
+     ;; Conjoin all values that pass the predicate condition
+     (if (pred x)
+       (conj output x)
+       output))
+   []
+   input))
+
+
+;; Implements update-in function
+(defn update-in-custom
+  [m k f]
+  ;; uses assoc-in but applies inputted function
+  (assoc-in
+   m k
+   (f (get-in m k))))
```

Comment:
OK - that one is a bit of a cheat... :-)

Can you do it without using get-in, or any of the *-in functions?  i.e. in terms
of more "primitive" functions?  Alternately, can you show me an implentation of get-in ?